pkg/v1/tkg/web/src/app/views/landing/azure-wizard/vnet-step/vnet-step.component.ts

```diff
@@ -273,11 +279,16 @@ export class VnetStepComponent extends StepFormDirective implements OnInit {
         let filteredSubnets = this.controlPlaneSubnets.filter(s => s.name === this.getSavedValue('controlPlaneSubnet', ''));
         if (filteredSubnets.length > 0) {
             this.formGroup.get('controlPlaneSubnet').setValue(filteredSubnets[0].cidr)
+        } else if (this.controlPlaneSubnets.length === 1) {
+            this.formGroup.get('controlPlaneSubnet').setValue(this.controlPlaneSubnets[0].name)
```

Comment:
nit: I would tend to start with the `if (this.controlPlaneSubnets.length === 1)` and put the filter and assignment in the `else` block.

Reply:
@swalner-vmware we have to make sure that there is at least one item in filteredSubnet array. otherwise filteredSubnets[0] will throw error.